inc/class-wp-sitemaps-renderer.php

```diff
@@ -198,12 +198,12 @@ public function get_sitemap_xml( $url_list ) {
 		foreach ( $url_list as $url_item ) {
 			$url = $urlset->addChild( 'url' );
 
-			// Add each attribute as a child node to the URL entry.
-			foreach ( $url_item as $attr => $value ) {
-				if ( 'url' === $attr ) {
-					$url->addChild( $attr, esc_url( $value ) );
-				} else {
-					$url->addChild( $attr, esc_attr( $value ) );
+			// Add each element as a child node to the URL entry.
+			foreach ( $url_item as $name => $value ) {
+				if ( 'loc' === $name ) {
+					$url->addChild( $name, esc_url( $value ) );
+				} elseif ( in_array( $name, array( 'lastmod', 'changefreq', 'priority' ), true ) ) {
+					$url->addChild( $name, esc_attr( $value ) );
 				}
```

Comment:
@pbiron to answer your question about `_doing_it_wrong`, I think if the wording is right we can add it.

e.g.

```php

// ...
} else {
	_doing_it_wrong(
		__FUNCTION__,
		/* translators: %s: list of element names */
		sprintf(
			__( 'Fields other than %s are not currently supported for sitemaps' ),
			implode( ',', array( 'loc', 'lastmod', 'changefreq', 'priority' ) )
		)
	)
}
```

Reply:
that's just what I was thinking.

Will there be any problem with that `_doing_it_wrong()` being removed in a future version when we do support extension elements?

Reply:
Not really

Reply:
will update the PR after the meeting